### PR TITLE
Move Binary_cross_entropy loss to Aten(CPU)

### PR DESCRIPTION
**Pull request by XiaobingSuper**
@VitalyFedyunin , this PR is about port binary_cross_entropy loss to Aten.
**Test script:**
```
import torch
import torch.nn as nn
import time

torch.manual_seed(1)

def _time():
    return time.time()

device = "cpu"
reduction = 'sum'
loss = nn.BCELoss(reduction=reduction)
m = nn.Sigmoid()

#warm up
for n in [10, 100, 1000, 10000]:
    input = m(torch.randn(128, n,device=device))
    input.requires_grad=True
    target = torch.randn(128, n, device=device)
    if reduction is 'none':
        grad = torch.randn(128, n, device=device)
    for i in range(1000):
        output = loss(input, target)
        if reduction is 'none':
            output.backward(grad)
        else:
             output.backward()

#get running time
for n in [10, 100, 1000, 10000]:
    fwd_t = 0
    bwd_t = 0
    input = m(torch.randn(128, n, device=device))
    input.requires_grad=True
    target = torch.randn(128, n, device=device)
    if reduction is 'none':
        grad = torch.randn(128, n, device=device)
    for i in range(10000):
        t1 = _time()
        output = loss(input, target)
        t2 = _time()
        if reduction is 'none':
            output.backward(grad)
        else:
            output.backward()
        t3 = _time()
        fwd_t = fwd_t + (t2 -t1)
        bwd_t = bwd_t + (t3 - t2)
    fwd_avg = fwd_t / 10000 * 1000
    bwd_avg = bwd_t / 10000 * 1000
    print("input size(128, %d) forward time is %.2f (ms); backwad avg time is %.2f (ms)."
          % (n, fwd_avg, bwd_avg))
```
Test Device: CPU: skx-8180
Perfromance:
Before:
```
reduction=’none’
input size(128, 10) forward time is 0.09 (ms); backwad avg time is 0.03 (ms).
input size(128, 100) forward time is 0.82 (ms); backwad avg time is 0.06 (ms).
input size(128, 1000) forward time is 8.19 (ms); backwad avg time is 0.46 (ms).
input size(128, 10000) forward time is 88.66 (ms); backwad avg time is 3.58 (ms).
reduction=’mean’
input size(128, 10) forward time is 0.09 (ms); backwad avg time is 0.04 (ms).
input size(128, 100) forward time is 0.80 (ms); backwad avg time is 0.15 (ms).
input size(128, 1000) forward time is 8.15 (ms); backwad avg time is 1.44 (ms).
input size(128, 10000) forward time is 85.45 (ms); backwad avg time is 13.81 (ms).
reduction=’sum’
input size(128, 10) forward time is 0.09 (ms); backwad avg time is 0.04 (ms).
input size(128, 100) forward time is 0.80 (ms); backwad avg time is 0.15 (ms).
input size(128, 1000) forward time is 8.18 (ms); backwad avg time is 1.45 (ms).
input size(128, 10000) forward time is 87.98 (ms); backwad avg time is 14.21 (ms).
```
After:
```
reduction=’none’
input size(128, 10) forward time is 0.02 (ms); backwad avg time is 0.03 (ms).
input size(128, 100) forward time is 0.03 (ms); backwad avg time is 0.04 (ms).
input size(128, 1000) forward time is 0.12 (ms); backwad avg time is 0.07 (ms).
input size(128, 10000) forward time is 1.07 (ms); backwad avg time is 0.14 (ms).
reduction=’mean’
input size(128, 10) forward time is 0.04 (ms); backwad avg time is 0.03 (ms).
input size(128, 100) forward time is 0.31 (ms); backwad avg time is 0.03 (ms).
input size(128, 1000) forward time is 2.99 (ms); backwad avg time is 0.14 (ms).
input size(128, 10000) forward time is 32.63 (ms); backwad avg time is 0.18 (ms)
reduction=’sum’
input size(128, 10) forward time is 0.04 (ms); backwad avg time is 0.03 (ms).
input size(128, 100) forward time is 0.31 (ms); backwad avg time is 0.03 (ms).
input size(128, 1000) forward time is 3.02 (ms); backwad avg time is 0.18 (ms).
input size(128, 10000) forward time is 32.64 (ms); backwad avg time is 0.14 (ms).
```
`OMP_NUM_THREADS=1:`
Before:
```
reduction=’none’
input size(128, 10) forward time is 0.09 (ms); backwad avg time is 0.03 (ms).
input size(128, 100) forward time is 0.79 (ms); backwad avg time is 0.06 (ms).
input size(128, 1000) forward time is 7.82 (ms); backwad avg time is 0.40 (ms).
input size(128, 10000) forward time is 77.60 (ms); backwad avg time is 3.70 (ms).
reduction=’mean’
input size(128, 10) forward time is 0.09 (ms); backwad avg time is 0.04 (ms).
input size(128, 100) forward time is 0.79 (ms); backwad avg time is 0.15 (ms).
input size(128, 1000) forward time is 7.80 (ms); backwad avg time is 1.31 (ms).
input size(128, 10000) forward time is 77.60 (ms); backwad avg time is 12.64 (ms).
reduction=’sum’
input size(128, 10) forward time is 0.09 (ms); backwad avg time is 0.04 (ms).
input size(128, 100) forward time is 0.79 (ms); backwad avg time is 0.15 (ms).
input size(128, 1000) forward time is 7.82 (ms); backwad avg time is 1.32 (ms).
input size(128, 10000) forward time is 77.88 (ms); backwad avg time is 12.68 (ms).
```
After:
```
reduction=’none’
input size(128, 10) forward time is 0.04 (ms); backwad avg time is 0.02 (ms).
input size(128, 100) forward time is 0.31 (ms); backwad avg time is 0.03 (ms).
input size(128, 1000) forward time is 2.85 (ms); backwad avg time is 0.18 (ms).
input size(128, 10000) forward time is 27.95 (ms); backwad avg time is 1.54 (ms).
reduction=’mean’
nput size(128, 10) forward time is 0.04 (ms); backwad avg time is 0.03 (ms).
input size(128, 100) forward time is 0.32 (ms); backwad avg time is 0.04 (ms).
input size(128, 1000) forward time is 2.91 (ms); backwad avg time is 0.17 (ms).
input size(128, 10000) forward time is 28.75 (ms); backwad avg time is 1.26 (ms).
reduction=’sum’
input size(128, 10) forward time is 0.04 (ms); backwad avg time is 0.03 (ms).
input size(128, 100) forward time is 0.33 (ms); backwad avg time is 0.04 (ms).
input size(128, 1000) forward time is 2.95 (ms); backwad avg time is 0.17 (ms).
input size(128, 10000) forward time is 28.64 (ms); backwad avg time is 1.25 (ms).
```

### CODE REVIEW

**Comment by XiaobingSuper** [comment]
After porting BCE loss, there will have a floating point arithmetic precision problem when using OMP to parallize the code, which will cause some test case failed, but those failed cases can be passed by set **OMP_NUM_THREADS** to **1**.

**Comment by XiaobingSuper** [comment]
#24682 Migrate binary_cross_entropy from the TH to Aten (CPU)
 #24683 Migrate binary_cross_entropy_backward from the TH to Aten (CPU)

**Comment by XiaobingSuper** [comment]
@VitalyFedyunin, there has a failed case about **test/cpp/api** , how do I reproduce it locally? Thanks!
